src/http-connection/query.codec.ts

```diff
@@ -197,11 +198,17 @@ class QuerySuccessResponseCodec extends QueryResponseCodec {
     }
 
     get meta(): Record<string, unknown> {
+        let profile = null
+        if (this._response.profiledQueryPlan != null) {
+            profile = this._decodeProfile(this._response.profiledQueryPlan)
+        }
+        else if (this._response.queryPlan != null) {
+            profile = this._decodeProfile(this._response.queryPlan)
+        }
         return {
             bookmark: this._response.bookmarks,
             stats: this._decodeStats(this._response.counters),
-            profile: this._response.profiledQueryPlan != null ?
-                this._decodeProfile(this._response.profiledQueryPlan) : null,
+            profile: profile,
```

Comment:
You might return the `queryPlan` only as `plan`. If you check the `ResultSummary` on drivers you will see that the `profile` can be return as `profile` and `plan`, but the `plan` is only returned as `plan` since it is a subset.